aicsimageio/readers/bioformats_reader.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+
+from functools import lru_cache
+from threading import Lock
+from typing import TYPE_CHECKING, Any, Dict, NamedTuple, Optional, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from fsspec.implementations.local import LocalFileSystem
+from ome_types import OME
+
+from .. import constants, dimensions, exceptions
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from ..utils.cached_property import cached_property
+from ..utils.dask_proxy import DaskArrayProxy
+from .reader import Reader
+
+if TYPE_CHECKING:
```

Comment:
I'm just curious, why are these only imported if TYPE_CHECKING?

Reply:
this again is just a habit (and I'm happy to move them all up to the top if it feels jarring to have this in some files and not in others.  apologies there).  The rationale is twofold:
1) i think putting things in TYPE_CHECKING is a nice, clear way to say "nothing here is needed at runtime... all of these imports will only appear in type annotations."
2) it helps to reduce the possibility of future circular imports.  that doesn't seem to be something that this library is in any danger any time soon, but in napari I've certainly had to go back and hide imports that had previously been imported at the tope just for the sake of type checking... hence the habit.

but like i said: i can see that it's a bit weird to have it sometimes but not always... happy to move it back up, shall i?

Reply:
understood.  I don't have a strong preference, but a mild one for moving it back up. Happy to let Jackson weigh in and don't want to just add meaningless busywork.

Reply:
I actually like the TYPE_CHECKING thing personally. It's definitely "safer" imo.

Reply:
Then I'm good with it.